### PR TITLE
Add preliminary support for aux camera calibration

### PR DESCRIPTION
**Pull request by mattalvarado**
Add initial support for the color aux camera calibration. This aux calibration data will need to written to flash in the firmware.

### CODE REVIEW

File: source/LibMultiSense/details/wire/SysCameraCalibrationMessage.h [hunk 1]
```diff
@@ -70,14 +70,14 @@ class CameraCalData {
...
-
-    //
-    // 2 MPix
+    static CRL_CONSTEXPR VersionType VERSION = 2;
 
     CameraCalData left;
     CameraCalData right;
 
+    CameraCalData aux;
+
```
Comment (Nyarmith) [inline source/LibMultiSense/details/wire/SysCameraCalibrationMessage.h:79]: too much whitespace?

File: source/Utilities/ImageCalUtility/ImageCalUtility.cc [hunk 2]
```diff
@@ -282,6 +288,12 @@ int main(int    argc,
         memcpy (&c.right.R[0][0], &data["R2"].front (), data["R2"].size () * sizeof (float));
         memcpy (&c.right.P[0][0], &data["P2"].front (), data["P2"].size () * sizeof (float));
 
+        memcpy (&c.aux.M[0][0], &data["M3"].front (), data["M3"].size () * sizeof (float));
+        memset (&c.aux.D[0], 0, sizeof (c.right.D));
```
Comment (Nyarmith) [inline source/Utilities/ImageCalUtility/ImageCalUtility.cc:292]: should it be `sizeof (c.aux.D)` ?

File: source/Utilities/ImageCalUtility/example_extrinsics.yml [hunk 3]
```diff
@@ -29,3 +29,19 @@ P2: !!opencv-matrix
    dt: d
    data: [ 5.9446551513671875e+02, 0., 512., -4.1660152435302734e+01, 0.,
        5.9446551513671875e+02, 520., 0., 0., 0., 1., 0. ]
+
```
Comment (Nyarmith) [inline source/Utilities/ImageCalUtility/example_extrinsics.yml:32]: extra newline?

File: source/Utilities/ImageCalUtility/example_intrinsics.yml [hunk 4]
```diff
@@ -25,3 +25,17 @@ D2: !!opencv-matrix
    data: [ -2.1223375201225281e-01, 1.3313443958759308e-01,
        -3.3149521914310753e-04, -4.0855954284779727e-04,
        -1.3206776231527328e-02, 0., 0., 0. ]
+
```
Comment (Nyarmith) [inline source/Utilities/ImageCalUtility/example_intrinsics.yml:28]: same

**Review by Nyarmith — CHANGES_REQUESTED** [review]

File: source/Utilities/ImageCalUtility/ImageCalUtility.cc (hunk 2 above)
Comment (mattalvarado) [reply source/Utilities/ImageCalUtility/ImageCalUtility.cc:292]: Good catch. Thanks

File: source/Utilities/ImageCalUtility/ImageCalUtility.cc [hunk 5]
```diff
@@ -87,6 +87,8 @@ std::ostream& writeImageIntrinics (std::ostream& stream, image::Calibration cons
     writeMatrix (stream, "D1", 1, 8, &calibration.left.D[0]);
     writeMatrix (stream, "M2", 3, 3, &calibration.right.M[0][0]);
     writeMatrix (stream, "D2", 1, 8, &calibration.right.D[0]);
+    writeMatrix (stream, "M3", 3, 3, &calibration.aux.M[0][0]);
```
Comment (psmithcrl) [inline source/Utilities/ImageCalUtility/ImageCalUtility.cc:90]: Do we need to check version here, what about uniots without aux imagers?

File: source/Utilities/ImageCalUtility/ImageCalUtility.cc [hunk 6]
```diff
@@ -97,6 +99,8 @@ std::ostream& writeImageExtrinics (std::ostream& stream, image::Calibration cons
     writeMatrix (stream, "P1", 3, 4, &calibration.left.P[0][0]);
     writeMatrix (stream, "R2", 3, 3, &calibration.right.R[0][0]);
     writeMatrix (stream, "P2", 3, 4, &calibration.right.P[0][0]);
+    writeMatrix (stream, "R3", 3, 3, &calibration.aux.R[0][0]);
```
Comment (psmithcrl) [inline source/Utilities/ImageCalUtility/ImageCalUtility.cc:102]: Do we need to check version here, what about units without aux imagers?

**Review by psmithcrl — COMMENTED** [review]
Just general questions about version for calibration, not sure if we need to add the explicit check for backwards compatibility.

**Review by psmithcrl — APPROVED** [review]

**Review by psmithcrl — CHANGES_REQUESTED** [review]
As I was going through the other side, I realized the version in SysGetCameraCalibrationMessage should also be updated to match else client won't be able to read the aux calibration.

**Comment by mattalvarado** [comment]
> As I was going through the other side, I realized the version in SysGetCameraCalibrationMessage should also be updated to match else client won't be able to read the aux calibration.

Not sure what you mean. That version should be independent of the `SysCameraCalibration` versioning. The firmware should just populate everything if it can.

**Comment by psmithcrl** [comment]
> > As I was going through the other side, I realized the version in SysGetCameraCalibrationMessage should also be updated to match else client won't be able to read the aux calibration.
> 
> Not sure what you mean. That version should be independent of the `SysCameraCalibration` versioning. The firmware should just populate everything if it can.



> > As I was going through the other side, I realized the version in SysGetCameraCalibrationMessage should also be updated to match else client won't be able to read the aux calibration.
> 
> Not sure what you mean. That version should be independent of the `SysCameraCalibration` versioning. The firmware should just populate everything if it can.

I see, sorry for the confusion.

**Review by psmithcrl — APPROVED** [review]

**Comment by mattalvarado** [comment]
@Nyarmith can you take another look at this and make sure it is good?

**Review by Nyarmith — APPROVED** [review]
LGTM

**Pull request merged**